### PR TITLE
New notification triggers: PUT/PATCH with 201 and 205 responses

### PR DESCRIPTION
**Pull request by CxRes**
This PR adds PUT and PATCH methods with 201 and 205 responses as notification triggers. 

External (Non-HTTP method) triggers will be dealt separately.

### CODE REVIEW

File: src/sections/triggers.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Triggers # {#triggers}
 
-The Solid Protocol [[!SOLID]] requires an LDP container's state is modified when certain HTTP events occur on a contained resource. Thus, an HTTP event on a resource can trigger a [=Solid server=] to transmit [PREP] notification not only on that resource but also on its container.
+The Solid Protocol [[!SOLID]] requires an LDP container's state is modified when certain HTTP events occur on a contained resource. Thus, an HTTP event on a resource can trigger a [=Solid server=] to transmit [PREP] notification on that resource or its container or both.
```
Comment (joachimvh) [inline src/sections/triggers.md:3]:
```suggestion
The Solid Protocol [[!SOLID]] requires an LDP container's state is modified when certain HTTP events occur on a contained resource. Thus, an HTTP event on a resource can trigger a [=Solid server=] to transmit a [PREP] notification on that resource or its container or both.
```

**Review by joachimvh — COMMENTED** [review]
I personally feel that the differentiation in all the different sections makes it more complicated to understand when notifications should be sent out.

Why not have 1 trigger section that states that if you do a PUT/PATCH/POST/DELETE and receive a 200/201/204/205 it can trigger a notification? The section below already details how creating or updating a resource can trigger specific notifications on the resource or its container.

**Comment by CxRes** [comment]
I want to be explicit about from **where** notifications are being sent out in each specific scenario (rather than have implementors work that out, which might lead to subtle variations between implementors). What I might propose in the spirit of your suggestion is to have one table with a third column saying "Notify on" (we could bikeshed that) with values of "resource", "container" and "resource and container". Would that be more readable?

**Comment by CxRes** [comment]
All this gives me an idea of how to rewrite the trigger section when I include non-HTTP triggers! So, I am thinking to make the unified table, merge this PR, and then I will seek your opinion after the rewrite on the whole section.

**Comment by CxRes** [comment]
Abandoned in favour of #4.

**Pull request closed without merging**